pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -410,6 +412,12 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 		kmsConfig:                config.KMSConfig,
 	}
 
+	// Register the provided KMS service with the plugin registry to allow using
+	// it for encryption of resources.
+	encryptionconfig.KMSPluginRegistry.Register(KMSServiceName, func(_ io.Reader) (envelope.Service, error) {
+		return gce.getGCPCloudKMSService()
```

Comment:
how frequently is a named plugin fetched (is it per decode/encode, or just once at init)? do we need to initialize this a single time or rate limit it?